0.63无法进入游戏，加载过程中发生卡死

GTNH 2.70 & 2.72客户端
TST 0.63
稳定触发

---- Minecraft Crash Report ----
// Don't be sad, have a hug! <3

Time: 2025-01-19 13:02:45 CST
Description: Initializing game

cpw.mods.fml.common.LoaderException: java.lang.NoClassDefFoundError: com.Nxer.TwistSpaceTechnology.common.machine.TST_ManufacturingCenter$ManufacturingMachineType
    at cpw.mods.fml.common.LoadController.transition(LoadController.java:163)
    at cpw.mods.fml.common.Loader.initializeMods(Loader.java:739)
    at cpw.mods.fml.client.FMLClientHandler.finishMinecraftLoading(FMLClientHandler.java:311)
    at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:552)
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:6099)
    at net.minecraft.client.main.Main.main(SourceFile:148)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:503)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:503)
    at oolloo.jlw.Wrapper.invokeMain(Wrapper.java:112)
    at oolloo.jlw.Wrapper.main(Wrapper.java:105)
Caused by: java.lang.NoClassDefFoundError: com.Nxer.TwistSpaceTechnology.common.machine.TST_ManufacturingCenter$ManufacturingMachineType
    at com.Nxer.TwistSpaceTechnology.common.machine.TST_ManufacturingCenter.<clinit>(TST_ManufacturingCenter.java:102)
    at com.Nxer.TwistSpaceTechnology.loader.MachineLoader.loadMachines(MachineLoader.java:739)
    at com.Nxer.TwistSpaceTechnology.TwistSpaceTechnology.init(TwistSpaceTechnology.java:106)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:503)
    at cpw.mods.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:532)
    at sun.reflect.GeneratedMethodAccessor13.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:503)
    at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
    at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
    at com.google.common.eventbus.EventBus.post(EventBus.java:275)
    at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:212)
    at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:190)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:503)
    at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
    at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
    at com.google.common.eventbus.EventBus.post(EventBus.java:275)
    at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:119)
    at cpw.mods.fml.common.Loader.initializeMods(Loader.java:737)
    ... 16 more
Caused by: java.lang.ClassNotFoundException: com.Nxer.TwistSpaceTechnology.common.machine.TST_ManufacturingCenter$ManufacturingMachineType
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191)
    at java.lang.ClassLoader.loadClassHelper(ClassLoader.java:959)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:904)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:887)
    ... 45 more
Caused by: java.lang.ClassFormatError: JVMCFRE113 意外的 EOF；类=com/Nxer/TwistSpaceTechnology/common/machine/TST_ManufacturingCenter$ManufacturingMachineType，偏移量=3421
    at java.lang.ClassLoader.defineClassImpl(Native Method)
    at java.lang.ClassLoader.defineClassInternal(ClassLoader.java:399)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:360)
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:182)
    ... 48 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Initialization --
Details:
Stacktrace:
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:6099)
    at net.minecraft.client.main.Main.main(SourceFile:148)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:503)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:503)
    at oolloo.jlw.Wrapper.invokeMain(Wrapper.java:112)
    at oolloo.jlw.Wrapper.main(Wrapper.java:105)
...
    Mixins in Stacktrace: 
        cpw.mods.fml.common.LoadController:
            mixin.mixinbooterlegacy.json:LoadControllerMixin from mod (unknown)
            mixins.gtnhmixins.json:LateMixinOrchestrationMixin from mod (unknown)
        net.minecraft.client.Minecraft:
            mixins.bettercrashes.early.json:MinecraftMixin from mod bettercrashes
            mixins.angelica.early.json:shaders.accessors.MixinMinecraft from mod angelica
            mixins.hodgepodge.early.json:minecraft.MixinMinecraft_ResizableFullscreen from mod hodgepodge
            mixins.archaicfix.early.json:client.lighting.MixinMinecraft from mod archaicfix
            mixins.angelica.early.json:sodium.MixinMinecraft from mod angelica
            mixins.modularui2.early.json:minecraft.MinecraftMixin from mod modularui2
            mixins.hodgepodge.early.json:minecraft.MixinMinecraft_UnfocusedFullscreen from mod hodgepodge
            mixins.archaicfix.early.json:client.core.MixinMinecraft from mod archaicfix
            mixins.hodgepodge.early.json:minecraft.MixinMinecraft_FastBlockPlacing from mod hodgepodge
            mixins.angelica.early.json:angelica.MixinMinecraft from mod angelica
            mixins.hodgepodge.early.json:minecraft.MixinMinecraft_UpdateKeys from mod hodgepodge
            mixins.hodgepodge.early.json:minecraft.MixinMinecraft_FixDuplicateSounds from mod hodgepodge
            mixins.hodgepodge.early.json:minecraft.profiler.MinecraftMixin from mod hodgepodge
            mixins.angelica.early.json:angelica.zoom.MixinMinecraft_Zoom from mod angelica
            mixins.dreamcraft.early.json:MixinMinecraft_PackIcon from mod dreamcraft
            mixins.dreamcraft.early.json:MixinMinecraft_ConfirmExit from mod dreamcraft
            mixins.hodgepodge.early.json:minecraft.MixinMinecraft_ClearRenderersWorldLeak from mod hodgepodge
            mixins.hodgepodge.early.json:minecraft.MixinMinecraft_ToggleDebugMessage from mod hodgepodge
        cpw.mods.fml.client.FMLClientHandler:
            mixins.angelica.early.json:angelica.MixinFMLClientHandler from mod angelica
            mixins.archaicfix.early.json:client.core.MixinFMLClientHandler from mod archaicfix
            mixins.hodgepodge.early.json:fml.MixinFMLClientHandler from mod hodgepodge
            mixins.hodgepodge.early.json:memory.MixinFMLClientHandler from mod hodgepodge
            mixins.angelica.early.json:angelica.archaic.MixinFMLClientHandler from mod angelica
            mixins.angelica.early.json:sodium.MixinFMLClientHandler from mod angelica
    Suspected Mods: Twist Space Technology (TwistSpaceTechnology)
    Launched Version: GTNH 2.7.0

--- a/src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/TST_ManufacturingCenter.java
+++ b/src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/TST_ManufacturingCenter.java
@@ -8,6 +8,7 @@ import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
+import java.util.Objects;
 
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
@@ -21,7 +22,6 @@ import org.jetbrains.annotations.NotNull;
 import com.Nxer.TwistSpaceTechnology.common.block.blockClass.Casings.multiuse.BlockMultiUseCore;
 import com.Nxer.TwistSpaceTechnology.util.TextEnums;
 import com.Nxer.TwistSpaceTechnology.util.TextWithColor;
-import com.github.bsideup.jabel.Desugar;
 import com.google.common.collect.BiMap;
 import com.google.common.collect.HashBiMap;
 import com.gtnewhorizon.structurelib.alignment.constructable.ISurvivalConstructable;
@@ -86,8 +86,17 @@ public class TST_ManufacturingCenter extends GTPPMultiBlockBase<TST_Manufacturin
 
     private static final BiMap<Integer, ManufacturingMachineType> MODES = HashBiMap.create();
 
-    @Desugar
-    private record ManufacturingMachineType(String typeName, RecipeMap<?> recipeMap, String unlocalizedName) {
+    private static final class ManufacturingMachineType {
+
+        private final String typeName;
+        private final RecipeMap<?> recipeMap;
+        private final String unlocalizedName;
+
+        private ManufacturingMachineType(String typeName, RecipeMap<?> recipeMap, String unlocalizedName) {
+            this.typeName = typeName;
+            this.recipeMap = recipeMap;
+            this.unlocalizedName = unlocalizedName;
+        }
 
         public static ManufacturingMachineType of(RecipeMap<?> recipeMap) {
             var recipeMapNameParts = recipeMap.unlocalizedName.split("\\.");
@@ -96,6 +105,46 @@ public class TST_ManufacturingCenter extends GTPPMultiBlockBase<TST_Manufacturin
                 recipeMap,
                 recipeMap.unlocalizedName);
         }
+
+        public String typeName() {
+            return typeName;
+        }
+
+        public RecipeMap<?> recipeMap() {
+            return recipeMap;
+        }
+
+        public String unlocalizedName() {
+            return unlocalizedName;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj == this) return true;
+            if (obj == null || obj.getClass() != this.getClass()) return false;
+            var that = (ManufacturingMachineType) obj;
+            return Objects.equals(this.typeName, that.typeName) && Objects.equals(this.recipeMap, that.recipeMap)
+                && Objects.equals(this.unlocalizedName, that.unlocalizedName);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(typeName, recipeMap, unlocalizedName);
+        }
+
+        @Override
+        public String toString() {
+            return "ManufacturingMachineType[" + "typeName="
+                + typeName
+                + ", "
+                + "recipeMap="
+                + recipeMap
+                + ", "
+                + "unlocalizedName="
+                + unlocalizedName
+                + ']';
+        }
+
     }
 
     static {

